src/core/services/theming/theming.js

```diff
@@ -492,6 +494,18 @@ function ThemingProvider($mdColorPalette) {
           el.data('$mdThemeController', ctrl);
         }
       }
+
+      /**
+       * Register change callback with parent mdTheme controller
+       */
+      function registerChangeCallback() {
+        var parentController = parent.controller('mdTheme');
```

Comment:
Then it should fall back to the watcher. I was trying to keep the logic the same, not improve upon it. That's why I didn't use `el.data('$mdThemeController')`. If you think I should use it, let me know and I can add it.